onnx/shape_inference/implementation.cc

```diff
@@ -236,107 +235,62 @@ std::string GetModelLocalFunctionsMapIdentifier(const std::string& domain, const
   return domain + ":" + func_name;
 }
 
-static void InferShapesImpl(
-    GraphProto* g,
-    const std::unordered_map<std::string, TypeProto*>& outer_scope_value_types_by_name,
-    const std::unordered_map<std::string, int>& opset_imports,
-    const ShapeInferenceOptions& options,
-    SymbolTable* symbol_table,
-    const ModelLocalFunctionsMap& model_local_functions_map,
-    const ISchemaRegistry* schema_registry = OpSchemaRegistry::Instance(),
-    const int ir_version = IR_VERSION // default the latest one
-) {
-  std::unordered_map<std::string, TypeProto*> value_types_by_name{outer_scope_value_types_by_name};
-  std::unordered_map<std::string, TypeProto*> undefined_value_types_by_name{outer_scope_value_types_by_name};
-  std::unordered_map<std::string, TensorShapeProto> generated_shape_data_by_name;
+// For graph-output types that are undefined, update them post-inference.
+// Q: Why exception is not saved as error if experimental op?
+// Document (in IR.md): domain for model-local functions
+// What is sparsetensortype.shape supposed to be?
```

Comment:
nit: do we want to keep this for future reference or it can be removed in this PR?

Reply:
Removed the comments, thanks!